File: apps/backend/src/EnterpriseAgentOs.Domain/features/Agents/IAgentPersonalityRepository.cs
namespace EnterpriseAgentOs.Domain.Features.Agents;

public interface IAgentPersonalityRepository
{
    Task<IReadOnlyList<AgentPersonalityRecord>> ListAsync(Guid agentId, CancellationToken ct = default);
    Task UpsertAsync(Guid agentId, string fileName, string content, CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/interfaces/IAgentPersonalityRepository.java
package com.enterpriseagentos.backendjava.domain.features.agents.interfaces;

import java.util.List;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

import com.enterpriseagentos.backendjava.domain.features.agents.models.AgentPersonalityModel;

public interface IAgentPersonalityRepository {
    CompletableFuture<List<AgentPersonalityModel>> listAsync(UUID agentId);
    CompletableFuture<Void> upsertAsync(UUID agentId, String fileName, String content);
}
